nfc/libs/NfcCoreLib/lib/Nci/phNciNfc_Discovery.c

```diff
@@ -1442,6 +1442,24 @@ static NFCSTATUS phNciNfc_DeActvNtfCb(void *pContext,void *pInfo,NFCSTATUS wStat
                         wDeActvStatus = NFCSTATUS_SUCCESS;
                     }
                 }
+                else if (pTransInfo->pbuffer[1] == (uint8_t)phNciNfc_e_DhRequestFailed)
+                {
+                    if ((pNciContext->NciDiscContext.eDeActvType == phNciNfc_e_SleepMode) && \
+                        (pTransInfo->pbuffer[0] == phNciNfc_e_SleepMode))
+                    {
+                        wDeActvStatus = NFCSTATUS_SUCCESS;
+                    }
+                    if ((pNciContext->NciDiscContext.eDeActvType == phNciNfc_e_SleepAfMode) && \
+                        (pTransInfo->pbuffer[0] == phNciNfc_e_SleepAfMode))
+                    {
+                        wDeActvStatus = NFCSTATUS_SUCCESS;
+                    }
+                    else if ((pNciContext->NciDiscContext.eDeActvType == phNciNfc_e_IdleMode) && \
+                             (pTransInfo->pbuffer[0] == phNciNfc_e_IdleMode))
```

Comment:
Conditions you implemented here look different from what is in previous scope on line 1426. There are more combinations of eDeActvType and pbuffer[0] values which would satisfy the confition - do you have any details on that?